### ISSUE TITLE
Brainstoriming Ideas for MCprep Branding

### ISSUE THREAD

**Issue by StandingPadAnimations** [Moo-Ack-Productions/MCprep#454]
MCprep's 10th anniversary is coming up soon, but MCprep doesn't really have any branding attached to it. Sure there is the Moo-Ack! Productions brand, but that's more of a wider umbrella brand and not really specific to MCprep. 

Some goals for branding:
- Simple, nothing to complex
- Consistent design
- Conveys an aspect of MCprep 

For example:
![MCprep-banner](https://github.com/Moo-Ack-Productions/MCprep/assets/75058058/03e65054-07d4-4266-800e-d267f00aa30b)

I went with a chest here since MCprep acts like a chest full of tools for Minecraft animation workflows. I made the background an off-white so that it doesn't end up being too stabby on people's eyes, and the font is IBM Plex Mono since it's a nice font (and open source).

We should hold a contest on the MCprep server for logo and branding designs based on these guidelines.

### Copyright and GPL compliance
Of course with MCprep's branding we should try to avoid copyright issues. For the example banner, I created a barebones style chest texture as to not infringe copyright by using barebones. In addition, to avoid copyright issues with the Minecraftia font, I just used IBM Plex Mono. If we do hold a contest for branding designs, we'll have to make sure that we aren't infringing on copyright.

As for copyright related to MCprep branding, we'd have to adhere to [GPL](https://web.archive.org/web/20230703230319/https://www.gnu.org/licenses/gpl-3.0.en.html), which states:
> Notwithstanding any other provision of this License, for material you add to a covered work, you may (if authorized by the copyright holders of that material) supplement the terms of this License with terms:
...
c) Prohibiting misrepresentation of the origin of that material, or requiring that modified versions of such material be marked in reasonable ways as different from the original version;
...
e) Declining to grant rights under trademark law for use of some trade names, trademarks, or service marks;

And in addition: 
> All other non-permissive additional terms are considered “further restrictions” within the meaning of section 10. If the Program as you received it, or any part of it, contains a notice stating that it is governed by this License along with a term that is a further restriction, you may remove that term. If a license document contains a further restriction but permits relicensing or conveying under this License, you may add to a covered work material governed by the terms of that license document, provided that the further restriction does not survive such relicensing or conveying.

To quote Section 10:
> You may not impose any further restrictions on the exercise of the rights granted or affirmed under this License. For example, you may not impose a license fee, royalty, or other charge for exercise of rights granted under this License, and you may not initiate litigation (including a cross-claim or counterclaim in a lawsuit) alleging that any patent claim is infringed by making, using, selling, offering for sale, or importing the Program or any portion of it.

To sum up what GPL says (based on my own interpretation):
- You can require modifications to be marked in a way that makes it clear they're modifications
- You can forbid branding use in modified and redistributed versions of the software for trademark reasons, or if there is a separate copyright license for said branding
- Beyond those restrictions, anything further would violate GPL

I think those are pretty reasonable terms, as long as we make sure to use a separate license for branding (I'm thinking something permissive would be fine).

**Comment by StandingPadAnimations** [comment]
@TheDuckCow just curious on your thoughts regarding this?

**Comment by SlincHD** [comment]
![McCreep](https://github.com/Moo-Ack-Productions/MCprep/assets/85231739/8b9bf94d-82a2-43b1-b00d-17cb263bb404)
your logo screamed for somthing to be better

**Comment by TheDuckCow** [comment]
Hey love the ideas here! Thanks both @StandingPadAnimations and @SlincHD for sharing some first stabs. This is an area where I'm not opposed to putting a couple $ towards a graphic designer. A contest seems like a good idea, but if we the core team aren't happy with the top pick, it's not a great idea to force using that for branding. But I do like the idea of getting community input here (and on discord and .. erm, X).

A chest is certainly something that nicely comes to mind with the analogy you described, though I'mo also not sure how well it stands out. I've also thought the "[happy leaping horse](https://theduckcow.com/wp-content/uploads/2016/08/mcstudio-horse.jpg)" was a fun motif, but maybe that's just me.

All the same though, I think asking for some commentary in a a thread on discord would be great. I'll get that started.

**Comment by CFeeney333** [comment]
How about these as something to work from as the logo? I think the third is the best personally. Any changes you would suggest? I have a couple other ideas i'll work on... (sry if the images are large)
![1](https://github.com/Moo-Ack-Productions/MCprep/assets/96747360/8abf6f00-12c4-4ad5-99ef-a30c9d1c0188)
![2](https://github.com/Moo-Ack-Productions/MCprep/assets/96747360/28627ca7-55bc-4325-a49b-89433369de4d)
![3](https://github.com/Moo-Ack-Productions/MCprep/assets/96747360/768a58a1-690f-460e-b987-aeac51be6809)

**Comment by StandingPadAnimations** [comment]
Works fine, but we can't use Minecraft textures for copyright reasons

**Comment by CFeeney333** [comment]
At all? Even though it's modified? How about using different texture pack?

**Comment by StandingPadAnimations** [comment]
> At all? Even though it's modified?

We don't want to deal with the copyright mess

> How about using different texture pack?

That's arguably worse since they're still copyrighted, but they're also from individuals (and people hate it when someone violates a small creator's copyright)

**Comment by TheDuckCow** [comment]
Hey @StandingPadAnimations we can leave at least the brainstorming to be open, in the end we will be able to create a design that is sufficiently different or customized from a given texture, even if it hints at it. Thanks @CFeeney333 for the brainstorming ideas! The more root idea I take from your design is something that shows off crafting in a way, but composed of multiple different parts or elements.

**Comment by StandingPadAnimations** [comment]
Regarding color scheme, I think the following 2 work well:
- #F5EFE9 
![image](https://github.com/Moo-Ack-Productions/MCprep/assets/75058058/f22f9f54-a8cd-4c5a-965f-0dcbeade8d09)

- 272727
![image](https://github.com/Moo-Ack-Productions/MCprep/assets/75058058/4491b028-e569-4d95-a21d-e9aabbbe9409)

They make for good light and dark theming and they contrast well but aren't too contrasting

**Comment by StandingPadAnimations** [comment]
I'm also currently looking for a nice light green color to complement the 2

**Comment by StandingPadAnimations** [comment]
Moving this for MCprep 4.0, as that will make for a good opportunity to introduce this

**Comment by Umaik101** [comment]
![MCPrep_Logo_v3](https://github.com/user-attachments/assets/c7ea174d-8e33-4fbe-b82d-b423ad5ad8ee)

Hey Dev-Team,

I was working on a logo design for MCprep and came up with this!
I took some inspiration from @StandingPadAnimations idea but instead of a chest I modeled a card box. 

- **Reason behind the card box**: Both a chest and a card box act in a similar way for they store stuff but a card box is usually used to move. So the idea behind the card box is that you take all the Minecraft stuff and "move" from one Minecraft related project in blender to another. MCprep would then be like a card box that stores all the Minecraft assets and more and that moves along with you.  

- **Player and Creeper**: These two miniature versions of Steve and a creeper represent the MC side of the add-on. The choice of using a creeper was actually rather arbitrary because I could have used any other mob instead. (But a touch of green actually does make it look more lively  :D)

- **Pickaxe**: It just represents the idea that with MCprep you "craft" your own stuff inside of blender.

- **Font**: Poppins (and btw. I used "v3.7.0" just as a kind of placeholder)

- **Textures**: I did not use any Minecraft textures. I eyeballed the colors of Steve and the creeper.

- **Process**: The logo was modeled and textured in Blockbench. I rendered it in Blender and edited it in Affinity Photo.  

I initially did the logo with only Steve but I thought it looked too simple and too boring. After I added the creeper I first couldn't decide between the two but I ended up liking the one with the creeper more :)
If there should be any copyright issues, let me know.

Hope you like it!

Umaik101

**Comment by TheDuckCow** [comment]
Just wanted to quickly chime in and say, amazing work @Umaik101 (and great to see you here). I do really like the direction this is going in. Let me think on it some and sample showing it in a few different treatments to see how it fares, but I would generally agree it has sufficient detail to not be boring, while still being distinct when smaller. Will also nudge community members to weigh in.

Thanks again! Oh, and maybe we can think about putting v4.0.0 as a placeholder, as they could be a nice time to unveil, depending on how the roadmap shakes out.

**Comment by Umaik101** [comment]
Thanks for the quick response @TheDuckCow and I am glad you like it!

I am of course open to any suggestion.

**Comment by StandingPadAnimations** [comment]
@Umaik101 Incredible design! My only issue with it are the positioning of the bottom text, it just seems really close to the stems of the P's in MCprep

**Comment by Umaik101** [comment]
https://github.com/user-attachments/assets/9d32c532-140d-46c7-b8b0-abdfa418982b

![MCPrep_Logo_Dark](https://github.com/user-attachments/assets/45b448c1-e936-4b68-b899-3376d7784560)
![MCPrep_Logo_Light](https://github.com/user-attachments/assets/fe66f724-6cc2-47ee-bce7-867e468de089)


![MCPrep_Logo_no_background ](https://github.com/user-attachments/assets/8abc4fe5-d7d4-4e2a-b5f9-75bbfe9d0fa4)
![MCPrep_Logo_no_background_dark](https://github.com/user-attachments/assets/0d33b956-40f9-43fb-a89a-e2bc8979f515)


Thanks @StandingPadAnimations, and noted that!

I took some time the last few days to refine the logo and to edit it a little bit more as the first design was kind of the general idea. This is what I now ended up with.

I also uploaded: 

- a video to show the different iterations of the design.
- the dark version
- the light version
- and the logo itself (for dark and light backgrounds)

Let me know if the edits are of your likings or if I should revert back to an older iteration.
